src/utils/detectionApi.ts

```diff
@@ -0,0 +1,506 @@
+/* eslint-disable max-classes-per-file */
+
+import {
+  CogniteClient,
+  EventSearch,
+  EventPatch,
+  DateRange,
+  CogniteInternalId,
+  IdEither,
+  Limit,
+  SinglePatchString,
+  SinglePatchDate,
+  ObjectPatch,
+  ArrayPatchLong,
+  Metadata,
+  RemoveField,
+  SetField,
+  Timestamp,
+  CogniteExternalId,
+  InternalId,
+  ExternalId,
+  EventChange,
+  ExternalEvent,
+  CogniteEvent,
+  EventFilterRequest,
+  ListResponse,
+  FilterQuery,
+  EventFilter,
+} from '@cognite/sdk';
+
+export enum ManualVerificationState {
+  UNHANDLED = 'unhandled',
+  VERIFIED = 'verified',
+  REJECTED = 'rejected',
+}
+
+export type SinglePatchManualVerificationState =
+  | SetField<ManualVerificationState>
+  | RemoveField;
+
+export interface BoundingBox2D {
+  left: number;
+  top: number;
+  width: number;
+  height: number;
+}
+
+export type SinglePatchBoundingBox2D = SetField<BoundingBox2D> | RemoveField;
+
+export interface ExternalDetection {
+  externalId?: CogniteExternalId;
+  startTime?: Timestamp;
+  endTime?: Timestamp;
+  label?: string;
+  description?: string;
+  metadata?: Metadata;
+  assetIds?: CogniteInternalId[];
+  box?: BoundingBox2D;
+  fileExternalId?: string;
+  manuallyVerified?: ManualVerificationState;
+}
+
+export interface CogniteDetection extends ExternalDetection, InternalId {
+  lastUpdatedTime: Date;
+  createdTime: Date;
+}
+
+export interface DetectionFilter {
+  startTime?: DateRange;
+  endTime?: DateRange;
+  createdTime?: DateRange;
+  lastUpdatedTime?: DateRange;
+  metadata?: Metadata;
+  /**
+   * Asset IDs of related equipment that this event relates to.
+   */
+  assetIds?: CogniteInternalId[];
+  /**
+   * The IDs of the root assets that the related assets should be children of.
+   */
+  rootAssetIds?: IdEither[];
+  /**
+   * Filter by label
+   */
+  label?: string;
+  /**
+   * Filter detections with an externalId starting with this value
+   */
+  externalIdPrefix?: string;
+  /**
+   * Filter based on file id
+   */
+  fileId?: string;
```

Comment:
Should this be `fileExternalId`? (cc: @runevalle-cognite )

Reply:
i added this
so the current logic is use externalId if exist, if not, then internalId

Reply:
i can change it to be fileExternalId instead :)